### PR TITLE
59 no snapshot missing snapshot

### PR DESCRIPTION
**Pull request by gvauvert**

### CODE REVIEW

File: iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java [hunk 1]
```diff
@@ -101,7 +102,9 @@ private String getFileName(long id, String storeName) {
                     .map(fileName -> {
                         Matcher matcher = m_filenamePattern.matcher(fileName);
                         if (!matcher.matches()) {
-                            System.err.println(fileName);
+                            throw new StoreException("Transaction file name in transaction directory does not match the expected pattern",
+                                                     args -> args.add("filename", fileName).add("transaction directory", m_transactionDir)
```
Comment (xfournet) [inline iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java]: args keys should be keys not sentences

File: iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java [hunk 2]
```diff
@@ -168,7 +171,7 @@ private long retrieveNextTxId() {
                     .filter(matcher -> matcher.matches() && TX_EXT.equals(matcher.group(3))) //
                     .mapToLong(matcher -> Long.valueOf(matcher.group(1))) //
                     .max() //
-                    .orElse(-1L) // in case no file exists we want nextTxId=0
+                    .orElse(-1L) // in case no file exists we want nextTxId=1 because a snapshot with no tx may have already be done to store the model
```
Comment (xfournet) [inline iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java]: i don't understand why the comment changed while the code is the same ?

File: iron-core/src/main/java/io/axway/iron/core/spi/file/IronMigration.java [hunk 3]
```diff
@@ -24,76 +30,133 @@
...
-                public FileVisitResult postVisitDirectory(Path dir, IOException exc) {
-                    return FileVisitResult.CONTINUE;
-                }
-            });
+            copyGlobalSnapshots(globalPath, targetGlobalPath);
+            copyTenantSnapshots(originIronPath, targetStoresPath);
+            completeStoreSnapshotWithMissingInstanceSnapshots(targetStoresPath);
 
-            System.out.println("Migration done to " + args[3]);
+            System.out.println("Migration done from " + originIronPath + " to " + targetIronPath);
```
Comment (xfournet) [inline iron-core/src/main/java/io/axway/iron/core/spi/file/IronMigration.java]: use printf instead string concatenation

File: iron-inttest/src/test/java/io/axway/iron/spi/file/FileStoreTest.java [hunk 4]
```diff
@@ -26,15 +27,18 @@
...
+import static java.util.stream.Collectors.*;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class FileStoreTest {
 
     @Test
     public void shouldCreateCompanySequenceBeRight() throws Exception {
         String randomStoreName = "iron-store-" + UUID.randomUUID();
-        Path filePath = Paths.get("tmp-iron-test", "iron-spi-file-inttest");
+        Path filePath = get("tmp-iron-test", "iron-spi-file-inttest");
```
Comment (xfournet) [inline iron-inttest/src/test/java/io/axway/iron/spi/file/FileStoreTest.java]: in this case, static import make the code less readable

File: iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java (hunk 1 above)
Comment (gvauvert) [reply iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java]: Fixed.

File: iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java (hunk 2 above)
Comment (gvauvert) [reply iron-core/src/main/java/io/axway/iron/core/spi/file/FileTransactionStore.java]: Code change reverted, not comment.
Fixed.

File: iron-core/src/main/java/io/axway/iron/core/spi/file/IronMigration.java (hunk 3 above)
Comment (gvauvert) [reply iron-core/src/main/java/io/axway/iron/core/spi/file/IronMigration.java]: Fixed.

File: iron-inttest/src/test/java/io/axway/iron/spi/file/FileStoreTest.java (hunk 4 above)
Comment (gvauvert) [reply iron-inttest/src/test/java/io/axway/iron/spi/file/FileStoreTest.java]: You're right.

**Pull request merged**